### ISSUE TITLE
Mini Cart block: improve button accessibility

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#4704]
Issue created from this comment by @opr:

> we should add some text to make the [Mini Cart] button more descriptive to screen readers. Right now my screen reader just says "9 items" with no further context about what the button is for.

I took some time to research what other websites are doing:
* Amazon: "1 item in shopping basket"
* Apple: "Shopping Bag"
* Asos: "Cart 1 item"
* eBay: "Your shopping cart contains 1 item"
* Glossier: "Cart Toggle"
* WayFair: "1 item in Cart"
* WooCommerce (Storefront mini cart): "10,00€ 1 item. View your shopping cart" (`View your shopping cart` comes from the [link title](https://github.com/woocommerce/storefront/blob/7074f6e063bb9cc285d08832a95377b7d2c0369e/inc/woocommerce/storefront-woocommerce-template-functions.php#L91)).

I think we can exclude anything like "View your shopping cart" or "Click to get more details", because screen readers will already announce that's a button, so it should be obvious. However, I think it's good to read the subtotal. So I would keep it like something like this:

"X items in cart, total price of Y"
